cli/internal/check/diff.go

```diff
@@ -17,6 +17,8 @@ import (
 	"github.com/invopop/jsonschema"
 )
 
+var callLLM = llm.CallLLM
```

Comment:
Rather than hoisting the function reference globally I think it would be cleaner to pass the function into Diff instead. That way we don't have to play with global variables.